Use of an internal okHttp package causes issues/crashes when using a more recent version of okHttp(4+).

https://github.com/DataDog/dd-sdk-android/blob/dde54aa25954bcdcaffb45b89a84d1ee5450e7ff/dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/net/RequestUniqueIdentifier.kt#L10

According to their own docs this is a little bit [naughty](https://square.github.io/okhttp/changelogs/upgrading_to_okhttp_4/#internal-api-changes)

Sure, FYI I'm using the `DatadogInterceptor`, `TracingInterceptor`, and `DatadogEventListener.Factory` as part of the `OkHttpClient.Builder`

but here is the exception:

```
E/AndroidRuntime: FATAL EXCEPTION: OkHttp Dispatcher
    Process: REDACTED, PID: 2258
    java.lang.NoClassDefFoundError: Failed resolution of: Lokhttp3/internal/Util;
        at com.datadog.android.core.internal.net.RequestUniqueIdentifierKt.identifyRequest(RequestUniqueIdentifier.kt:19)
        at com.datadog.android.DatadogEventListener$Factory.create(DatadogEventListener.kt:231)
        at okhttp3.internal.connection.RealCall.<init>(RealCall.kt:70)
        at okhttp3.OkHttpClient.newCall(OkHttpClient.kt:281)

Caused by: java.lang.ClassNotFoundException: Didn't find class "okhttp3.internal.Util" on path: DexPathList[[zip file REDACTED ]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:259)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        at com.datadog.android.core.internal.net.RequestUniqueIdentifierKt.identifyRequest(RequestUniqueIdentifier.kt:19) 
        at com.datadog.android.DatadogEventListener$Factory.create(DatadogEventListener.kt:231) 
```

I've removed some sensitive parts.

Version of okHttp we use is: **4.9.3**

--- a/dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/net/RequestUniqueIdentifier.kt
+++ b/dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/net/RequestUniqueIdentifier.kt
@@ -7,7 +7,7 @@
 package com.datadog.android.core.internal.net
 
 import okhttp3.Request
-import okhttp3.internal.Util
+import java.io.IOException
 
 /**
  * Generates an identifier to uniquely track requests.
@@ -16,11 +16,25 @@ internal fun identifyRequest(request: Request): String {
     val method = request.method()
     val url = request.url()
     val body = request.body()
-    return if (body == null || body == Util.EMPTY_REQUEST) {
+    return if (body == null) {
         "$method•$url"
     } else {
-        val contentLength = body.contentLength()
+        val contentLength = try {
+            body.contentLength()
+        } catch (@Suppress("SwallowedException") ioe: IOException) {
+            0
+        }
         val contentType = body.contentType()
-        "$method•$url•$contentLength•$contentType"
+        // TODO RUMM-3062 It is possible that if requests are say GZIPed (as an example), or maybe
+        //  streaming case (?), they all will have contentLength = -1, so if they target the same URL
+        //  they are going to have same identifier, messing up reporting.
+        //  -1 is valid return value for contentLength() call according to the docs and stands
+        //  for "unknown" case.
+        //  We need to have a more precise identification.
+        if (contentType != null || contentLength != 0L) {
+            "$method•$url•$contentLength•$contentType"
+        } else {
+            "$method•$url"
+        }
     }
 }

--- a/dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/net/RequestUniqueIdentifierTest.kt
+++ b/dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/net/RequestUniqueIdentifierTest.kt
@@ -13,13 +13,17 @@ import fr.xgouchet.elmyr.junit5.ForgeExtension
 import okhttp3.MediaType
 import okhttp3.Request
 import okhttp3.RequestBody
+import okio.BufferedSink
 import org.assertj.core.api.Assertions.assertThat
 import org.junit.jupiter.api.Test
 import org.junit.jupiter.api.extension.ExtendWith
 import org.junit.jupiter.api.extension.Extensions
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ValueSource
 import org.mockito.junit.jupiter.MockitoExtension
 import org.mockito.junit.jupiter.MockitoSettings
 import org.mockito.quality.Strictness
+import java.io.IOException
 
 @Extensions(
     ExtendWith(MockitoExtension::class),
@@ -171,5 +175,41 @@ internal class RequestUniqueIdentifierTest {
         val contentLength = fakeBody.length
         assertThat(id)
             .isEqualTo("DELETE•$fakeUrl•$contentLength•$fakeContentType; charset=utf-8")
+    }
+
+    @ValueSource(strings = ["POST", "PUT", "PATCH", "DELETE"])
+    @ParameterizedTest
+    fun `identify request { body#contentLength throws exception }`(
+        method: String
+    ) {
+        val body = object : RequestBody() {
+            override fun contentLength(): Long {
+                throw IOException("")
+            }
+
+            override fun contentType(): MediaType? {
+                return MediaType.parse(fakeContentType)
+            }
+
+            override fun writeTo(sink: BufferedSink) {
+                // no-op
+            }
+        }
+        val request = Request.Builder()
+            .apply {
+                when (method) {
+                    "POST" -> post(body)
+                    "PUT" -> put(body)
+                    "PATCH" -> patch(body)
+                    "DELETE" -> delete(body)
+                }
+            }
+            .url(fakeUrl)
+            .build()
+
+        val id = identifyRequest(request)
+
+        assertThat(id)
+            .isEqualTo("$method•$fakeUrl•0•$fakeContentType")
     }
 }

